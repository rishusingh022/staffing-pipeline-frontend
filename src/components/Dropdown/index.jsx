import React, { useState } from 'react';
import { BiChevronDown } from 'react-icons/bi';
import { AiOutlineSearch } from 'react-icons/ai';
import PropTypes from 'prop-types';

export default function Dropdown({ dropdownName, dropdownData, selectOption }) {
  const [inputValue, setInputValue] = useState('');
  const [selected, setSelected] = useState('');
  const [toggleDropdown, setToggleDropdown] = useState(false);

  return (
    <div className="w-44 font-normal">
      <div
        onClick={() => setToggleDropdown(!toggleDropdown)}
        className="bg-white w-full p-2 flex justify-between items-center border-solid border border-black cursor-pointer">
        {selected ? selected : `${dropdownName}`}
        <BiChevronDown size={20} />
      </div>
      <div className={`${!toggleDropdown ? 'hidden' : ''}`}>
        <ul className={'bg-white mt-2 overflow-y-auto max-h-60 border border-black-300'}>
          <div className="flex items-center sticky top-0 h-10 bg-white">
            <AiOutlineSearch size={18} className="w-10" />
            <input
              type="text"
              placeholder="search"
              value={inputValue}
              onChange={e => setInputValue(e.target.value.toLowerCase())}
              className="placeholder:text-gray-700 outline-none text-mg w-36 pr-2"
            />
          </div>

          {dropdownData.map(option => (
            <li
              key={option}
              className={`p-2 text-sm hover:bg-gray-300 cursor-pointer
            ${option.toLowerCase().startsWith(inputValue) ? 'block' : 'hidden'}`}
              onClick={() => {
                if (option.toLowerCase() !== selected.toLowerCase()) {
                  setSelected(option);
                  selectOption(option);
                }
                setToggleDropdown(false);
              }}>
              {option}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}
Dropdown.propTypes = {
  dropdownName: PropTypes.string.isRequired,
  dropdownData: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectOption: PropTypes.func.isRequired,
};
