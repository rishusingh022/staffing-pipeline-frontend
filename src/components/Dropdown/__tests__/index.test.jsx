import React from 'react';
import { render } from '@testing-library/react';
import Dropdown from '..';

describe('Dropdown', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Dropdown dropdownName="technology" dropdownData={['react', 'nodejs', 'express']} selectOption={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
