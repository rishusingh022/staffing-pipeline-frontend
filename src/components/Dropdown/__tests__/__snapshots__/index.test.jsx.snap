// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
<DocumentFragment>
  <div
    class="w-44 font-normal h-80 m-16"
  >
    <div
      class="bg-white w-full p-2 flex justify-between items-center border-solid border border-black cursor-pointer"
    >
      technology
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
        />
      </svg>
    </div>
    <ul
      class="bg-white mt-2 overflow-y-auto max-h-0"
    >
      <div
        class="flex items-center sticky top-0 h-10 bg-white"
      >
        <svg
          class="w-10"
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
        <input
          class="placeholder:text-gray-700 outline-none text-mg w-36 pr-2"
          placeholder="search"
          type="text"
          value=""
        />
      </div>
      <li
        class="p-2 text-sm hover:bg-gray-300 hover:text-white cursor-pointer
            block"
      >
        react
      </li>
      <li
        class="p-2 text-sm hover:bg-gray-300 hover:text-white cursor-pointer
            block"
      >
        nodejs
      </li>
      <li
        class="p-2 text-sm hover:bg-gray-300 hover:text-white cursor-pointer
            block"
      >
        express
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
